src/js/components/Select/SelectContainer.js

```diff
@@ -403,7 +442,11 @@ const SelectContainer = forwardRef(
                 step={theme.select.step}
                 onMore={onMore}
                 replace={replace}
-                show={activeIndex !== -1 ? activeIndex : undefined}
+                show={
+                  ![-1, options.length].includes(activeIndex)
```

Comment:
The `show` prop was responsible for undesirable auto-scroll to the end of the list of options when focus was applied to the clear button in bottom position.

Like the search input, it should be exempt from the behaviour.